core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -600,28 +600,48 @@ private[spark] class ExecutorAllocationManager(
    * the executor is not already marked as idle.
    */
   private def onExecutorIdle(executorId: String): Unit = synchronized {
-    if (executorIds.contains(executorId)) {
-      if (!removeTimes.contains(executorId) && !executorsPendingToRemove.contains(executorId)) {
-        // Note that it is not necessary to query the executors since all the cached
-        // blocks we are concerned with are reported to the driver. Note that this
-        // does not include broadcast blocks.
-        val hasCachedBlocks = blockManagerMaster.hasCachedBlocks(executorId)
-        val now = clock.getTimeMillis()
-        val timeout = {
-          if (hasCachedBlocks) {
-            // Use a different timeout if the executor has cached blocks.
-            now + cachedExecutorIdleTimeoutS * 1000
-          } else {
-            now + executorIdleTimeoutS * 1000
+    onExecutorIdle(Seq(executorId))
+  }
+
+  private def onExecutorIdle(executorIds: Seq[String], onUnpersist: Boolean = false): Unit =
+    synchronized {
+      executorIds.foreach(executorId => {
+        if (executorIds.contains(executorId)) {
```

Comment:
These two lines seem redundant; by definition the seocnd line is true because of the new one before it.